server/tests/test_v1.py

```diff
@@ -401,3 +401,23 @@ def test_agents_post_bad(mongo_app):
 
     assert 422 == output.status_code
     assert "Validation error" in output.text
+
+
+def test_get_agents_data(mongo_app):
+    """Test api to retrieve agent data"""
+    app, _ = mongo_app
+    agent_name = "agent1"
+    agent_data = {
+        "state": "provision",
+        "queues": ["q1", "q2"],
+        "location": "here",
+    }
+    output = app.post(f"/v1/agents/data/{agent_name}", json=agent_data)
+    assert 200 == output.status_code
```

Comment:
I've typically shied away from setting up tests through other app code. So I'd create objects in db directly instead of calling post endpoints. As I felt that would be more of a unit test. Frankly, I'm not sure if there is really any benefit to that. From you're experience and knowledge, have I been wasting my time :sweat_smile: ?

Reply:
I don't have a strong opinion here, since we have other tests that test posting that data also. I feel like using the API is a little more realistic than direct manipulation of the db - when possible. But I've done both, as sometimes it's not easy to inject the data you actually want, because you need to simulate a very specific condition.